src/Events/Category_Colors/Migration/Handler.php

```diff
@@ -0,0 +1,145 @@
+	 *
+	 * @since TBD
+	 *
+	 * @var bool
+	 */
+	protected bool $dry_run = false;
+
+	/**
+	 * Determines whether the migration step is in a valid state to run.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool True if the migration step can run, false otherwise.
+	 */
+	public function is_runnable(): bool {
+		return true;
+	}
+
+	/**
+	 * Runs the migration process in sequential steps.
+	 *
+	 * Each step is executed in order, stopping if an error occurs.
+	 * This ensures that the migration does not continue if a previous step fails.
+	 *
+	 * @since TBD
+	 *
+	 * @param bool $dry_run Optional. If true, the migration runs in dry-run mode.
+	 *
+	 * @return bool Returns true on success, false on failure.
+	 */
+	public function process( bool $dry_run = false ): bool {
+		$this->dry_run = $dry_run;
+		$start_time    = microtime( true );
+
+		if ( Status::$postprocess_completed === static::get_migration_status()['status'] ) {
+			$this->log_elapsed_time( 'Migration has already been completed.', $start_time );
+
+			return true;
+		}
+
+		// Prevent running if migration is already in progress.
+		if ( Status::$execution_in_progress === static::get_migration_status()['status'] ) {
+			$this->log_elapsed_time( 'Migration is already in progress.', $start_time );
+
+			return false;
+		}
+
+		$this->log_message( 'info', 'Migration starting. Current status: ' . static::get_migration_status()['status'] );
+
+		// Define migration steps.
+		$migration_steps = [
+			'Preprocessing'   => fn() => tribe( Pre_Processor::class )->process(),
+			'Validation'      => fn() => tribe( Validator::class )->process(),
+			'Execution'       => fn() => tribe( Worker::class )->process(),
+			'Post-processing' => fn() => tribe( Post_Processor::class )->process(),
+		];
+
+		// Execute each step in sequence and stop if a failure occurs.
+		foreach ( $migration_steps as $step_name => $step ) {
+			$error = $this->run_migration_step( $step, $step_name );
+
+			if ( is_wp_error( $error ) ) {
+				$error_message = is_wp_error( $error ) ? $error->get_error_message() : 'Unknown error.';
+				$this->log_elapsed_time( "Migration failed at step: {$step_name}. Error: {$error_message}", $start_time );
+
+				return false;
+			}
+		}
+
+		$this->log_elapsed_time( 'Migration Process', $start_time );
+
+		return true;
+	}
+
+	/**
+	 * Runs a single migration step and stops further execution if an error occurs.
+	 *
+	 * @since TBD
+	 *
+	 * @param callable $migration_step The function representing a migration step.
+	 * @param string   $step_name      The name of the migration step.
+	 *
+	 * @return true|WP_Error Returns true on success, WP_Error on failure.
+	 */
+	protected function run_migration_step( callable $migration_step, string $step_name ) {
+		$result = $migration_step();
+
+		if ( is_wp_error( $result ) || false === $result ) {
+			return $this->log_message( 'error', "Migration failed at step: {$step_name}. Stopping further processing." );
+		}
+
+		return true;
+	}
+}
```

Comment:
type hinting the return type is powerful for identifing bugs! When you need to do something like that i would suggest to throw Exceptions.

Reply:
This logic/message is taken directly from the WP Core https://github.com/WordPress/WordPress/blob/master/wp-includes/taxonomy.php#L1471

You are unable to update meta on a shared Term.